src/components/ApplyFiltersModal/state.ts

```diff
@@ -0,0 +1,46 @@
+import { useReducer } from "react";
+import { Filter } from "types/logs";
+
+type Action =
+  | { type: "ADD_FILTER"; filterToAdd: Filter }
+  | { type: "REMOVE_FILTER"; filterToRemove: string }
+  | { type: "RESET" };
+
+type State = {
+  selectedFilters: Filter[];
+};
+
+const initialState = (): State => ({
+  selectedFilters: [],
+});
```

Comment:
I think a plain object suffices here
```suggestion
const initialState: State = {
  selectedFilters: [],
};
```

Reply:
That's really interesting, seems like a cool way to get around this is actually this [optional 3rd argument](https://react.dev/reference/react/useReducer#parameters) known as the initializer function! I thought [this writeup](https://mtm.dev/reset-usereducer-state) was helpful.

Reply:
thanks for explaining, I didn't know this 🙏